internal/gateway/domain_name_instance.go

```diff
@@ -67,130 +67,9 @@ func domainNameInstanceDeleted(
-	}
-
-	// update workload resource instance
-	_, err = client.UpdateWorkloadResourceInstance(r.APIClient, r.APIServer, virtualService)
-	if err != nil {
-		if errors.Is(err, client.ErrorObjectNotFound) {
-			// workload resource instance has already been deleted
-			return 0, nil
-		}
-		log.Error(err, "failed to create workload resource instance")
-		return 0, nil
-	}
-
-	// trigger a reconciliation of the workload instance
-	workloadInstanceReconciled := false
-	workloadInstance.Reconciled = &workloadInstanceReconciled
-	_, err = client.UpdateWorkloadInstance(r.APIClient, r.APIServer, workloadInstance)
-	if err != nil {
-		if errors.Is(err, client.ErrorObjectNotFound) {
-			// workload instance has already been deleted
-			return 0, nil
-		}
-		log.Error(err, "failed to update workload instance")
-		return 0, nil
-	}
-
 	return 0, nil
 }
 
-// reconcileCreatedOrUpdatedDomainNameInstance performs reconciliation when a
-// domain name instance has been created or updated.
-func reconcileCreatedOrUpdatedDomainNameInstance(
-	r *controller.Reconciler,
-	domainNameInstance *v0.DomainNameInstance,
-	log *logr.Logger,
-) error {
-
-	// validate threeport state
-	err := validateThreeportStateExternalDns(r, domainNameInstance, log)
-	if err != nil {
-		return fmt.Errorf("failed to validate threeport state: %w", err)
-	}
-
-	// get workload instance
-	workloadInstance, err := client.GetWorkloadInstanceByID(r.APIClient, r.APIServer, *domainNameInstance.WorkloadInstanceID)
-	if err != nil {
-		return fmt.Errorf("failed to get workload instance: %w", err)
-	}
-
-	// get domain name definition
-	domainNameDefinition, err := client.GetDomainNameDefinitionByID(r.APIClient, r.APIServer, *domainNameInstance.DomainNameDefinitionID)
-	if err != nil {
-		return fmt.Errorf("failed to get domain name definition: %w", err)
-	}
-
-	// configure virtual service
-	workloadResourceInstance, err := configureWorkloadResourceInstance(r, domainNameDefinition, workloadInstance)
-	if err != nil {
-		return fmt.Errorf("failed to configure virtual service: %w", err)
-	}
-
-	// update workload resource instance
-	_, err = client.UpdateWorkloadResourceInstance(r.APIClient, r.APIServer, workloadResourceInstance)
-	if err != nil {
-		return fmt.Errorf("failed to create workload resource instance: %w", err)
-	}
-
-	// trigger a reconciliation of the workload instance
-	workloadInstanceReconciled := false
-	workloadInstance.Reconciled = &workloadInstanceReconciled
-	_, err = client.UpdateWorkloadInstance(r.APIClient, r.APIServer, workloadInstance)
-	if err != nil {
-		return fmt.Errorf("failed to update workload instance: %w", err)
-	}
-
-	// mark domain name instance as reconciled
-	domainNameInstanceReconciled := true
-	domainNameInstance.Reconciled = &domainNameInstanceReconciled
-	_, err = client.UpdateDomainNameInstance(r.APIClient, r.APIServer, domainNameInstance)
-	if err != nil {
-		return fmt.Errorf("failed to update domain name instance: %w", err)
-	}
-
-	return nil
-}
```

Comment:
Previously, this reconciler would update a `VirtualService` object with the correct domain. While implementing the `SubDomain`field on the `Gateway` object, I realized that reconciler can simply refer to the domain from a `DomainNameDefinition` and construct a full domain (either subdomain + domain, or just domain) based off the `GatewayInstance` subdomain + related `DomainNameDefinition` domain. As a result, this logic can be removed from this reconciler.